src/App.js

```diff
@@ -1,29 +1,34 @@
-import logo from './logo.svg';
+import React from 'react';
 import './App.css';
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit
-          {' '}
-          <code>src/App.js</code>
-          {' '}
-          and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
-}
+import Layout from './components/Layout';
+import Investigator from './components/Investigator';
+import Main from './components/Main';
+import Registration from './components/Registration';
+import SignUp from './components/SignUp';
+import Investigators from './components/Investigators';
+import InvestigatorForm from './components/InvestigatorForm';
+import ReserveForm from './components/ReserveForm';
+import DeleteInvestigator from './components/DeleteInvestigator';
+import Reservations from './components/Reservations';
+
+const App = () => (
+  <BrowserRouter>
+    <Layout>
+      <Routes>
+        <Route path="/" element={<Registration />} />
+        <Route path="/main" element={<Main />} />
+        <Route path="/signup" element={<SignUp />} />
+        <Route path="/investigators" element={<Investigators />} />
+        <Route path="/investigators/:id" element={<Investigator />} />
+        <Route path="/add_investigator" element={<InvestigatorForm />} />
+        <Route path="/investigators/:id/reserve" element={<ReserveForm />} />
+        <Route path="remove_investigator" element={<DeleteInvestigator />} />
```

Comment:
I don't think we really need to have a component or route for deleting items. We just have to utilize the endpoint from the backend for the action.